hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -576,6 +580,10 @@ public void initialize(URI name, Configuration originalConf)
 
       s3aInternals = createS3AInternals();
 
+      s3ObjectStorageClassFilter = Optional.ofNullable(conf.get(READ_RESTORED_GLACIER_OBJECTS))
```

Comment:
use getTrimmed(key, ""), toUpper() and then do the matching. fail meaningfully if the value isn't recognised.